File: ios/seniorityApp/Bulb.m
//
//  Bulb.m
//  seniorityApp
//
//  Created by Dawid Urbas on 22/03/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(Bulb, NSObject)
RCT_EXTERN_METHOD(turnOn)
RCT_EXTERN_METHOD(turnOff)
RCT_EXTERN_METHOD(getStatus: (RCTResponseSenderBlock)callback)
@end

File: ios/Bulb.swift
//
//  Bulb.swift
//  seniorityApp
//
//  Created by Dawid Urbas on 22/03/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

@objc(Bulb)class Bulb: NSObject {

@objc static var isOn = false
  
@objc static func requiresMainQueueSetup() -> Bool {
  return true
}

@objc func turnOn() {
  Bulb.isOn = true
  print("Bulb is now ON")
}
  
@objc func turnOff() {
  Bulb.isOn = false
  print("Bulb is now OFF")
}
  
@objc func getStatus(_ callback: RCTResponseSenderBlock) {
  callback([NSNull(), Bulb.isOn])
}
}
